charts/redpanda/values.yaml

```diff
@@ -252,6 +271,120 @@ storage:
     labels: {}
     # Additional annotations to apply to the created PersistentVolumeClaims.
     annotations: {}
+  # Tiered Storage helps to lower storage costs by offloading log segments to cloud storage.
+  # You can specify the amount of local storage you want to retain in local storage.
+  # You don't need to specify which log segments you want to move because Redpanda moves them automatically
+  # based on cluster-level configuration properties. Redpanda Tiered Storage indexes where data is offloaded,
+  # so it can retrieve the data when you need it.
+  #
+  # REF: https://docs.redpanda.com/docs/manage/tiered-storage/#tiered-storage-configuration-properties
+  #
+  # Topic configuration flags override any cluster-level configuration. For example, on new topics,
+  # if `cloud_storage_enable_remote_write` is set to `true`, you can set `redpanda.remote.write` to `false`
+  # to turn it off for a particular topic. Or, if you have Tiered Storage turned off for the cluster, you can enable it
+  # for individual topics with the topic flags. When cluster-level properties are changed,
+  # the changes apply only to new topics, not existing topics.
+  #
+  # REF: https://docs.redpanda.com/docs/manage/tiered-storage/#enable-tiered-storage-for-a-topic
+  tieredConfig:
+    # Global flag that enables Tiered Storage feature if license is provided.
+    # Default is false.
+    cloud_storage_enabled: false
+    # Cluster wide config option to enable data upload from Redpanda to cloud storage if set to true.
+    cloud_storage_enable_remote_write: true
+    # Cluster wide config option to enable Redpanda to fetch data from cloud storage if set to true.
+    cloud_storage_enable_remote_read: true
+    # This configuration option is available starting from 22.3.X
+    #
+    # Possible values:
+    # `config` - If IAM roles are not available, specify credentials in the cluster configuration file.
+    # `aws_instance_metadata` - For an AWS EC2 instance, use the instance metadata API from AWS.
```

Comment:
I'm concerned about all this copy-paste.  How is this going to stay up to date when we evolve the configuration options, e.g. adding new possible values to this one?

If we decide as a team that maintaining a copy of the docs in these .yaml files makes sense then it's doable, but it needs thought, and some kind of process similar to how the docs team tracks new features and makes updates for them.

Reply:
Agreed. We shouldn't embed config docs within Helm but instead link straight to the platform docs for description.

Reply:
> I'm concerned about all this copy-paste. How is this going to stay up to date when we evolve the configuration options, e.g. adding new possible values to this one?

Once the Redpanda make official release with the appropriate docs update, then K8S team would pick the new Redpanda version and adjust comment with the Redpanda container tag (Redpanda version).

> If we decide as a team that maintaining a copy of the docs in these .yaml files makes sense then it's doable, but it needs thought, and some kind of process similar to how the docs team tracks new features and makes updates for them.

Release notes, from the Redpanda release, should drive changes in this values.yaml file.

> Agreed. We shouldn't embed config docs within Helm but instead link straight to the platform docs for description.

If users start from [artifact hub](https://artifacthub.io/packages/helm/redpanda-data/redpanda), then it add friction as user would need to jump into many docs to correctly configure Redpanda cluster. This is only tiered storage section, but currently we have 3 different link to our docs. This PR will add 3 more.

I'm used to have this `value.yaml` definition included in artifact hub similar to https://artifacthub.io/packages/helm/softonic/kyverno#values

We don't generate such table of content, but we could use [values modal](https://artifacthub.io/packages/helm/redpanda-data/redpanda?modal=values) to show what is available to customers. That way we can include as many description as we need. 

Another option is to create automation that would generate [values.yaml](https://github.com/redpanda-data/helm-charts/blob/main/charts/redpanda/values.yaml) from the [values.schema.json](https://github.com/redpanda-data/helm-charts/blob/main/charts/redpanda/values.schema.json), but each field needs to have description. You can see in [values schema modal](https://artifacthub.io/packages/helm/redpanda-data/redpanda?modal=values-schema) that not many fields have description. As per https://github.com/redpanda-data/helm-charts/pull/278#discussion_r1072799550 comment.

I'm trying to show you that there would be friction for users if they are discovering Redpanda from artifiact hub.

-------

To add more I included constraint for this particular filed in https://github.com/redpanda-data/helm-charts/pull/278/files#diff-7573c8e3ada7cb2fe4a9a6b40e135a1d5568a9d128d15a57b124d32568126819R368.

That way user would know the `cloud_storage_credentials_source` field have only 4 available options. The command `helm install` or `helm upgrade` will reject invalid input. It's much better user experience.

Example:
```
helm template redpanda charts/redpanda --set storage.tieredConfig.cloud_storage_enabled=true --set image.tag=v22.1.1 --set license_key=test --set storage.tieredConfig.cloud_storage_credentials_source=test
Error: values don't meet the specifications of the schema(s) in the following chart(s):
redpanda:
- storage.tieredConfig.cloud_storage_credentials_source: Does not match pattern '^(config|aws_instance_metadata|sts|gcp_instance_metadata)$'
```

If we are making some effort to add input validation, why we should not follow up with clear description?

Reply:
Yes having a well documented values.yaml is [considered best practice](https://helm.sh/docs/chart_best_practices/values/#document-valuesyaml) within the helm community. A poorly documented values.yaml, or a link to external documentation will create an endless stream if issue requests asking for it to be documented. This has been tried in helm charts for other popular software and did not go over well.

Reply:
This thread came up [here](https://github.com/redpanda-data/helm-charts/pull/278#discussion_r1081344609). I don't see a huge problem with copying in details that can already be found in our docs, nor with linking to docs. I'm fine with either of these two options.

But at least one of them is needed, as users need to have the information available in `values.yaml` to successfully deploy Redpanda, as this file could likely be the first source of information on deploying Redpanda that some users have.

Reply:
I think having a well-defined `values.yaml` will become more important in the future once Redpanda readily available in other environments like OpenShift.

Reply:
Please bear in mind that the core team's release notes are not intended to be at a level of detail that would enable accurately maintaining a copy of docs for tunables.  Picking an example from this PR, if we changed cloud_storage_initial_backoff_ms's default for some reason, I would probably not release note that.

I think we're likely to get a higher quality outcome by using the json schema to do an update of this stuff at each major feature release, rather than sifting it out of release notes and updating by hand.  Ultimately the docs on the website will work this way too, for things like keeping the defaults etc in sync.

Reply:
I think I understand what you're suggeting, @jcsp, but would you please create an issue so we can track that?